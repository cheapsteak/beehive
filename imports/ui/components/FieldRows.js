// Framework
import React from "react";

// Components
import { Container, Row, Col, Form } from "reactstrap";
import FieldRow from "./FieldRow";

/**
 * A list of field rows with a header.
 *
 * @param {Object} props All the properties passed to the React Component.
 * @returns {Element} Stateless functional React component.
 */
const FieldRows = ({ heading = "Fields", fields = {}, stateSettingAction }) => (
  <Container className="field-rows px-0" fluid>
    <Row noGutters>
      <Col>
        <h3>{heading}</h3>
      </Col>
    </Row>
    <Row noGutters>
      <Col>
        <Form>
          {Object.keys(fields).map((label, i) => (
            <FieldRow
              key={i}
              label={label}
              value={fields[label]}
              stateSettingAction={stateSettingAction}
              visibleLabel
            />
          ))}
        </Form>
      </Col>
    </Row>
  </Container>
);

export default FieldRows;
