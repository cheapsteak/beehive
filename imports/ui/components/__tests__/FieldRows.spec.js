import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import FieldRows from '../FieldRows';

describe('<FieldRows />', () => {
  let defaultProps;

  describe('FieldRows SnapShot', () => {
    beforeEach(() => {
      defaultProps = {
        heading: 'Fields',
        fields: {
          tickrate: 50
        },
        stateSettingAction: jest.fn()
      };
    });

    it('renders FieldRows correctly with default props', () => {
      delete defaultProps.heading;
      delete defaultProps.fields;
      const tree = ReactTestRenderer.create(
        <FieldRows {...defaultProps} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders FieldRows correctly', () => {
      const tree = ReactTestRenderer.create(
        <FieldRows {...defaultProps} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
