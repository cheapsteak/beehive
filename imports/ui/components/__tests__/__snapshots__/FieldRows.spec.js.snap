// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldRows /> FieldRows SnapShot renders FieldRows correctly 1`] = `
<div
  className="field-rows px-0 container-fluid"
>
  <div
    className="no-gutters row"
  >
    <div
      className="col"
    >
      <h3>
        Fields
      </h3>
    </div>
  </div>
  <div
    className="no-gutters row"
  >
    <div
      className="col"
    >
      <form
        className=""
      >
        <div
          className="field-row form-group"
        >
          <label
            className=""
            htmlFor="tickrate-number-field-row-input"
          >
            Tickrate
          </label>
          <input
            className="form-control"
            id="tickrate-number-field-row-input"
            name="tickrate-number-field-row-input"
            onChange={[Function]}
            placeholder="Tickrate"
            type="text"
            value={50}
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`<FieldRows /> FieldRows SnapShot renders FieldRows correctly with default props 1`] = `
<div
  className="field-rows px-0 container-fluid"
>
  <div
    className="no-gutters row"
  >
    <div
      className="col"
    >
      <h3>
        Fields
      </h3>
    </div>
  </div>
  <div
    className="no-gutters row"
  >
    <div
      className="col"
    >
      <form
        className=""
      />
    </div>
  </div>
</div>
`;
